src/mlpack/methods/preprocess/preprocess_imputer_main.cpp

```diff
@@ -162,7 +161,33 @@ static void mlpackMain()
 
       for (size_t i : dirtyDimensions)
       {
-        imputer.Impute(input, missingValue, i);
+        if (strategy == "mean")
+        {
+          Imputer<double, MapperType, MeanImputation<double>> imputer(info);
+          imputer.Impute(input, missingValue, i);
```

Comment:
This works, but I think it might simplify the code a little if you included the loop over `dirtyDimensions` inside the if statements, i.e.

```
          Imputer<double, MapperType, MeanImputation<double>> imputer(info);
          for (size_t i : dirtyDimensions)
            imputer.Impute(input, missingValue, i);
```

What do you think?

Reply:
Yes, that's definitely a better way. It will reduce unnecessary repeative checks.